### PR TITLE
Add string support for data check action code

### LINKED ISSUE #1904: Make data check action codes accessible via strings
Following up from @freddyaboulton's suggestion [here](https://github.com/alteryx/evalml/pull/1896#discussion_r583770941), making data check action codes a string representation similar to what we have for problem types would make it easier for users to handle and manipulate.

### PR DESCRIPTION
**Pull request by bchen1116**
fix #1904

### CODE REVIEW

File: evalml/data_checks/utils.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        ValueError: If input is not a string or DatCheckActionCode object.
+
+    Examples:
+        >>> assert handle_data_check_action_code("drop_col") == DataCheckActionCode.DROP_COL
+        >>> assert handle_data_check_action_code("DROP_ROWS") == DataCheckActionCode.DROP_ROWS
+        >>> assert handle_data_check_action_code("Impute_col") == DataCheckActionCode.IMPUTE_COL
+    """
+    if isinstance(action_code, str):
+        try:
+            tpe = DataCheckActionCode._all_values[action_code.upper()]
```
Comment (angela97lin) [inline evalml/data_checks/utils.py:25]: jw, what does tpe stand for? :o

File: evalml/tests/data_checks_tests/test_data_check_action.py [hunk 2]
```diff
@@ -158,3 +158,33 @@ def test_convert_dict_to_action(dummy_data_check_name):
...
+
+@pytest.mark.parametrize(
+    "action_code,expected_code",
+    [
+        ("drop_rows", DataCheckActionCode.DROP_ROWS),
+        ("Drop_col", DataCheckActionCode.DROP_COL),
+        ("TRANSFORM_TARGET", DataCheckActionCode.TRANSFORM_TARGET),
+    ],
+)
+def test_data_check_action_equality_string_input(
```
Comment (angela97lin) [inline evalml/tests/data_checks_tests/test_data_check_action.py:171]: nitpick but maybe we want to update the names `action_code` and `expected_code`? Seems like what we're trying to test is that the stringified version is equivalent to the enum, so something like `action_code_string` vs 'action_code`?

File: evalml/tests/data_checks_tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        ValueError,
+        match="`handle_data_check_action_code` was not passed a str or DataCheckActionCode object",
+    ):
+        handle_data_check_action_code(None)
+
+    with pytest.raises(
+        ValueError,
+        match="`handle_data_check_action_code` was not passed a str or DataCheckActionCode object",
+    ):
+        handle_data_check_action_code(ProblemTypes.BINARY)
```
Comment (angela97lin) [inline evalml/tests/data_checks_tests/test_utils.py:22]: 😂 
Nice

**Review by angela97lin — APPROVED** [review]
LGTM, thank you for taking this on! 🙏  Just left a few really nit-picky comments :)

**Review by jeremyliweishih — APPROVED** [review]
noice!

File: evalml/data_checks/utils.py (hunk 1 above)
Comment (bchen1116) [reply evalml/data_checks/utils.py:25]: left over from previous code 😅 shall change!

**Pull request merged**